public/app/plugins/datasource/loki/components/LokiQueryFieldForm.tsx

```diff
@@ -164,7 +164,7 @@ export class LokiQueryFieldForm extends React.PureComponent<LokiQueryFieldFormPr
               {chooserText}
             </ButtonCascader>
           </div>
-          <div className="gf-form gf-form--grow flex-shrink-1">
+          <div className={'gf-form gf-form--grow flex-shrink-1 min-width-15 explore-input-margin'}>
             <QueryField
```

Comment:
Add min-width so QueryField and doesn't shrink to nothing. `explore-input-margin` adds right margin to query editors that have extra field.